homeassistant/components/hydrawise/coordinator.py

```diff
@@ -40,8 +43,31 @@ async def _async_update_data(self) -> HydrawiseData:
         user = await self.api.get_user()
         controllers = {}
         zones = {}
+        sensors = {}
+        daily_water_use: dict[int, ControllerWaterUseSummary] = {}
         for controller in user.controllers:
             controllers[controller.id] = controller
             for zone in controller.zones:
                 zones[zone.id] = zone
-        return HydrawiseData(user=user, controllers=controllers, zones=zones)
+            for sensor in controller.sensors:
+                sensors[sensor.id] = sensor
+
+            if any(
```

Comment:
Add an empty `ControllerWaterUseSummary` for the controller before this branch. That way there's always one populated for the controller, even if there's no sensor. Then you don't have to call `.get()` when fetching it from the dict later.

Reply:
Fixed.